11.8.6 Hotfix

jwoodruff40 opened the pull request:
## What type of PR is this?
- [x] Bug
- [ ] Change
- [ ] Feature
- [ ] Miscellaneous

<details><summary><i><b>
Sub-categories:
</b></i></summary>

- [ ] Template
- [ ] Map
- [ ] Config
- [ ] Function
- [ ] Localization

</details>

## What have you changed, and why?

**Information:**

> - fix garage access limit for air vehicles on dedicated / non-host clients
> - fix bad (old?) vest classnames in 3CBF ION arid faction
> - don't charge players for rebuilding assets unless assets were actually rebuilt (fixes https://discord.com/channels/817005365740044289/817014691092103228/1452083797083291749)
> - revert erroneous change to Chernarus Autumn mission.sqm
> - allow AI units to equip any primary weapon from initialRebelEquipment if they have no class-specific primaries available. [e.g. let medics have a bolt action rifle (usually categorized as a sniper rifle) if they have no SMGs / carbines available). Should fix issues with new games with very limited starting equipment.
> - move extender params back to their own section in the params dropdown so they go in the right place instead of being appended under the development params
> - change default for limiting weapons by AI class to off / no
> - Add param to default custom loadouts to legacy behavior; that is, all tabs / items start as overridden / saved, and you can choose to not save (randomize) specific tabs / items. Default for the param is yes / on: legacy behavior
> - re-enable flag actions for static weapons (enable/disable AI, move static). The option to move statics was accidentally dropped in 376/718, this re-adds it by using the "static" case in fn_flagaction again.

**How can your changes be tested, or reproduced?**

> ...

**Does this PR include changes not authored by you?**

- [x] No
- [ ] Yes (See below)

<details><summary><i><b>
Further info:
</b></i></summary><br>

- [ ] I confirm that I, and by extension this repository, can legally use these third-party changes. (Provide links or author attribution.)

> ...

</details>

## Please verify the following (If possible).

`*` - Mandatory

- [x] These changes are my own, or I have written permission to use them. `*`
- [x] These changes are ready for review, or will be marked as a draft. `*`
- [x] I have loaded the mission in LAN host.
- [x] I have loaded the mission on a dedicated server.

Is further work needed?

- [ ] Needs further testing.
- [ ] Needs further changes.
- [ ] Needs to be converted to a draft.

## Please specify which Issue this PR Resolves (If Applicable).
This PR closes #724

<hr><details><summary><i><b>
Notes:
</b></i></summary><br>

> ...

</details>

jwoodruff40 committed:
Fix garage access limit on dedicated;

publicVariable A3A_buildingsToSave

jwoodruff40 committed:
bad classnames in 3CBF ION faction

jwoodruff40 committed:
don't charge players for rebuilding assets unless assets were actually rebuilt;

add message for nothing to repair;

de-duplicate re-used code in fn_rebuildAssets

UnseenKill commented on A3A/addons/core/functions/Base/fn_rebuildAssets.sqf:
```diff
@@ -61,30 +73,27 @@ switch (true) do {
 	default {
 		private _militaryBuildings = nearestObjects [_position, A3A_buildingWhitelist, 500,  true];
 
+		private _destroyedCount = count destroyedBuildings; diag_log _destroyedCount;
```
Ixnay on the `diag_log`

jwoodruff40 committed:
Update fn_rebuildAssets.sqf

jwoodruff40 committed:
revert erroneous change to Chernarus Autumn mission.sqm

jwoodruff40 committed:
allow non-specialized AI classes to use sniper rifles

jwoodruff40 committed:
allow specialized AI units to equip any primary from initialRebelEquipment if they have no class-specific primaries available

jwoodruff40 committed:
move extender params back to their own section in params dropdown

jwoodruff40 committed:
change default for limiting weapons by AI class to no

jwoodruff40 committed:
re-enable flagactions for static weapons (enable/disable AI, move static)

jwoodruff40 committed:
Add param to default custom loadouts to legacy behavior;

that is, all tabs / items start as overridden / saved, and you can choose to not save (randomize) specific tabs / items.

Default for the param is yes / on: legacy behavior

jwoodruff40 committed:
update version to 11.8.6

jwoodruff40 committed:
rework rebuild fix + DS compat

Silence-112 reviewed (approved):
![di-goose-honk](https://github.com/user-attachments/assets/c5f781f3-ce63-45de-8372-df2c2f4bbafe)
